tests/integration/kubernetes/gha-run.sh

```diff
@@ -121,6 +121,11 @@ function cleanup() {
     echo "Gather information about the nodes and pods before cleaning up the node"
     get_nodes_and_pods_info "yes"
 
+    if [ "${platform}" = "aks" ]; then
+        delete_cluster
+        return
+    fi
```

Comment:
Please, allow kata-deploy to be uninstalled from aks clusters, so we can know whether this actually working or not.

If you want to add a short-circuit to go directly to the cliuster deletion, I'm fine, but in the CI runs I'd prefer not changing the folow.

Reply:
Ouch, it seems this is needed as without it we've been leaving pods behind.

I am approving it, and later on we can improve it if needed.

Reply:
> Please, allow kata-deploy to be uninstalled from aks clusters, so we can know whether this actually working or not.

This makes sense, hadn't thought about this.

> Ouch, it seems this is needed as without it we've been leaving pods behind.

Don't understand this one actually. Can you explain? Aren't we fine if we delete the cluster in the end?

Reply:
It seems we're failing in a cleanup step and we're never triggering the cluster delete part. :-/